internal/configs/module_call.go

```diff
@@ -169,7 +169,18 @@ func (mc *ModuleCall) decodeStaticSource(eval *StaticEvaluator) hcl.Diagnostics
 		// NOTE: This code was originally executed as part of decodeModuleBlock and is now deferred until we have the config merged and static context built
 		var err error
 		if mc.VersionAttr != nil {
-			mc.SourceAddr, err = addrs.ParseModuleSourceRegistry(mc.SourceAddrRaw)
+			// Check if version is null before forcing registry module parsing
+			val, valDiags := eval.Evaluate(mc.VersionAttr.Expr, StaticIdentifier{
```

Comment:
If you re-order the decode functions above in decodeStaticFields, we could remove this additional logic as you already set mc.VersionAttr to nil.